src/realm/util/logger.cpp

```diff
@@ -175,10 +176,14 @@ const std::string_view Logger::level_to_string(Level level) noexcept
     return "";
 }
 
+StderrLogger::StderrLogger() noexcept
+    : m_mutex(s_stderr_logger_mutex)
+{
+}
+
 void StderrLogger::do_log(const LogCategory& cat, Level level, const std::string& message)
 {
-    static Mutex mutex;
-    LockGuard l(mutex);
+    std::lock_guard l(m_mutex);
```

Comment:
Is there a reason we need to capture the global mutex as a member variable reference? Could the whole change for this ticket just be changing this line from `static Mutex mutex;` to `static auto& mutex = *new Mutex;` ? Or changing the mutex from a Mutex to a std::mutex as you do above and locking it here without changing any of the headers?

Reply:
I wasn't sure if there was a risk that the `static auto& mutex` variable could become invalid so I was saving a local reference to the mutex. Fortunately, this should be defined in the `.data` region of the application, so actually there shouldn't be any risk of it going away. Just verified with a simple test, so I'll remove the extra reference.

Reply:
Also verified that the minimal set of logger changes are work fine with the Dart SDK.

Reply:
Can you expand a bit on how the variable could become invalid and how taking a reference to it would prevent it from becoming invalid? Do you mean there was a risk the heap-allocated mutex could go out-of-scope and be destroyed without anything calling delete on it? How would taking a reference to it prevent that and not just lead to a dangling reference? How does being in the `.data` section of the image change things here?

Reply:
I was referring to the global static reference (`s_stderr_logger_mutex`) variable being destroyed/going away during the app exit. But, since globals/statics like this are symbols defined in the program memory (`.bss` or `.data` sections), it will not go away during the lifetime of the app.